backend/app/routers/options.py

```
@@ -84,6 +90,38 @@ def scan_options(
+    for the relax math.
+
+    Errors (generic strings only, per CLAUDE.md):
+
+    - 400 ``"rule not relaxable"`` — binary or unknown rule family.
+    - 500 ``"Failed to compute relax preview"`` — unexpected internal failure.
+    """
+    try:
+        rules = load_rules_config(db)
+        return preview_relax(req, rules)
+    except NotRelaxableError:
+        raise HTTPException(status_code=400, detail="rule not relaxable")
```

Ruff (0.15.13)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)